clang/lib/Sema/SemaExpr.cpp

```diff
@@ -3107,8 +3107,12 @@ Sema::PerformObjectMemberConversion(Expr *From,
                                    /*IgnoreAccess=*/true))
     return ExprError();
 
-  return ImpCastExprToType(From, DestType, CK_UncheckedDerivedToBase,
-                           VK, &BasePath);
+  Qualifiers FromTypeQuals = FromType.getQualifiers();
```

Comment:
Maybe add a quote from https://eel.is/c++draft/basic.type.qualifier#def:object,volatile

Reply:
Added, thanks!